### PR TITLE
Hw12 andriy povzun

### PR DESCRIPTION
**Pull request by AndriyPovzun**

### CODE REVIEW

File: Andriy_Povzun/HW12/Task1.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    def add_new_employees(self, quantity_new_employees: int):
+        self.__number_employees = self.__number_employees + quantity_new_employees
+
+    @number_employees.deleter
+    def number_employees(self):
+        pass
+
+    """
+    Deleter written so that users do not break the design of the class
+    """
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: docstring should be before pass

File: Andriy_Povzun/HW12/Task1.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    The method is used to change the company name
+    """
+
+    @company_name.deleter
+    def company_name(self):
+        pass
+
+    """
+    Deleter written so that users do not break the design of the class
+    """
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: docstring should be before pass

File: Andriy_Povzun/HW12/Task1.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    @property
+    def company_products(self):
+        return self.__company_products
+
+    def add_new_products(self, *args: str):
+        self.__company_products.append(*args)
+
+    """
+    The method is used to add new company products
+    """
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: doc string should be inside the method. Please fix it in all places

File: Andriy_Povzun/HW12/Task1.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    def withdraw_from_production(self, *args: str):
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: *args is not a string. It is tuple

File: Andriy_Povzun/HW12/Task1.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    """
+    Deleter written so that users do not break the design of the class
+    """
+
+    @property
+    def company_products(self):
+        return self.__company_products
+
+    def add_new_products(self, *args: str):
+        self.__company_products.append(*args)
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: Looks like a company can have the same products. Please fix it

File: Andriy_Povzun/HW12/Task1.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    """
+    Deleter written so that users do not break the design of the class
+    """
+
+    @property
+    def company_products(self):
+        return self.__company_products
+
+    def add_new_products(self, *args: str):
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py]: *args is not a string

File: Andriy_Povzun/HW12/Task2.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+        self.__position = position
+        self.__salary = salary
+
+    @property
+    def name(self):
+        return self.__name
+
+    """
+    The method is used to obtain the name of the employee
+    """
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task2.py]: same here. please fix it in all places

File: Andriy_Povzun/HW12/Task2.py [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    def name(self):
+        pass
+
+    """
+    Deleter written so that users do not break the design of the class
+    """
+
+    @property
+    def surname(self):
+        return self.__surname
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task2.py:40]: the surname can be changed as well. When change passport or get married

File: Andriy_Povzun/HW12/Task2.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    def surname(self):
+        pass
+
+    """
+    Deleter written so that users do not break the design of the class
+    """
+
+    @property
+    def age(self):
+        return self.__age
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task2.py:63]: you can implement grow_up method

File: Andriy_Povzun/HW12/Task1.py (hunk 1 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: Done.

File: Andriy_Povzun/HW12/Task1.py (hunk 2 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: Done.

File: Andriy_Povzun/HW12/Task1.py (hunk 3 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: Done.

File: Andriy_Povzun/HW12/Task1.py (hunk 4 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: fix.

File: Andriy_Povzun/HW12/Task1.py (hunk 5 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: fix.

File: Andriy_Povzun/HW12/Task1.py (hunk 6 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py]: okay, changed.

File: Andriy_Povzun/HW12/Task2.py (hunk 7 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task2.py]: done.

File: Andriy_Povzun/HW12/Task2.py (hunk 8 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task2.py:40]: add function.

File: Andriy_Povzun/HW12/Task2.py (hunk 9 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task2.py:63]: add, thanks.

File: Andriy_Povzun/HW12/Task1.py [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+    def add_new_products(self, *args: tuple):
+        """
+           The method is used to add new company products
+        """
+        for arg in args:
+            if arg in self.__company_products:
+                raise TypeError(f'Such a product already exists: {arg}')
+            else:
+                self.__company_products.append(arg)
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task1.py:50]: it can be done much easier. just think about it. Without any errors

File: Andriy_Povzun/HW12/Task2.py [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+        """
+        The method is used to obtain the age of the employee
+        """
+        return self.__age
+
+    def grow_up(self, ages: int):
+        """
+        The employee will become older by the number of years indicated in the function
+        """
+        self.__age += ages
```
Comment (bohdandovbsyh) [inline Andriy_Povzun/HW12/Task2.py]: what if I pass -10 to parameters?

Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task2.py]: fixed.

File: Andriy_Povzun/HW12/Task1.py (hunk 10 above)
Comment (AndriyPovzun) [reply Andriy_Povzun/HW12/Task1.py:50]: Done.

Comment (bohdandovbsyh) [reply Andriy_Povzun/HW12/Task1.py:50]: Okay. I mead set type of data. But let it be

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**